src/BadRequestError.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Copyright 2023 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use Exception;
+
+/**
+ * BadRequestError indicates a malformatted request.
```

Comment:
I wouldn't mind a little more detail here
```suggestion
 * BadRequestError indicates a malformatted request. This is thrown internally
 * when a protobuf message fails to be deserialized in the request body.
```